lumicks/pylake/force_calibration/power_spectrum_calibration.py

```diff
@@ -305,18 +345,19 @@ def fit_power_spectrum(
     chi_squared_per_deg = chi_squared / n_degrees_of_freedom
     backing = (1 - scipy.special.gammainc(chi_squared / 2, n_degrees_of_freedom / 2)) * 100
 
-    # Fitted power spectrum values.
-    ps_model = power_spectrum.with_spectrum(
-        model(power_spectrum.frequency, *solution_params), power_spectrum.num_points_per_block
-    )
-
     # When using theoretical weights for fitting, ref [5] mentions that the found value for D will
-    # be biased by a factor (n+1)/n. Multiplying by n/(n+1) compensates for this.
+    # be biased by a factor (n+1)/n. Multiplying by n/(n+1) compensates for this. Note that it's
+    # strictly correct only for loss_function == "gaussian".
     if bias_correction:
         bias_corr = power_spectrum.num_points_per_block / (power_spectrum.num_points_per_block + 1)
         solution_params[1] *= bias_corr
         perr[1] *= bias_corr
 
+    # Fitted power spectrum values.
+    ps_model = power_spectrum.with_spectrum(
+        model(power_spectrum.frequency, *solution_params), power_spectrum.num_points_per_block
+    )
```

Comment:
I did switch it because of that. I get what you're saying, but as a user I would find it weird to get a different curve back from the fit than the curve I get when I plug the reported fit parameters into the same model again. Also, the bias is deterministic and proven to be there, so was it really the "best" fit? My opinion doesn't matter too much, but I would have chosen to return a curve the user would also reconstruct based on the fit results.

Reply:
Yeah, it's a fair point. I do agree. If we plot _with_ the bias correction, the user can actually assess the impact of it on the fit, while now they can't (though I doubt they'd even notice to be honest).

I would shelve this for the next breaking (major) release since it is a breaking change. People tend to read release notes better on major releases.

Reply:
Yeah that works for me, I'll swap it back for now. Do we keep track of it with a ticket?

Reply:
Yeah, I added it to the considerations for `2.0` ticket. When we start gearing up for that one, I'll bring it up again.